plugins/pulp_rpm/plugins/distributors/yum/metadata/package.py

```diff
@@ -29,42 +29,38 @@ def __init__(self, working_dir, checksum_type=None):
         """
 
         metadata_file_path = os.path.join(working_dir, REPO_DATA_DIR_NAME, PACKAGE_XML_FILE_NAME)
-        super(PackageXMLFileContext, self).__init__(metadata_file_path, checksum_type)
+        super(PackageXMLFileContext, self).__init__(metadata_file_path, 'comps',
+                                                    checksum_type=checksum_type)
 
-    def _write_root_tag_open(self):
+    def _open_metadata_file_handle(self):
         """
-        Write the opening tag for the root element of a given metadata XML file.
+        Open the metadata file handle, creating any missing parent directories.
+        If the file already exists, this will overwrite it.
         """
+        super(XmlFileContext, self)._open_metadata_file_handle()
+        self.xml_generator = XMLWriter(self.metadata_file_handle, short_empty_elements=True)
```

Comment:
I only know this because I also was recently working with sax, but I think the ``short_empty_elements`` is a new parameter in python 3.2. I'm not sure how it worked for you.

https://docs.python.org/3/library/xml.sax.utils.html#xml.sax.saxutils.XMLGenerator

What do you think? I know the docs for sax are missing a lot of stuff, so maybe it's actually available in python 2?